lib/logstash/outputs/elasticsearch/template_manager.rb

```diff
@@ -12,12 +12,10 @@ def self.install_template(plugin)
     end
 
     private
-    def self.get_es_version(client)
-      client.get_version
-    end
-
     def self.get_es_major_version(client)
-      get_es_version(client)["number"][0]
+      # get the elasticsearch version of each node in the pool and
+      # pick the biggest major version
+      client.connected_es_versions.uniq.map {|version| version.split(".").first.to_i}.sort.last
```

Comment:
s/sort.last/max

there was already a PR in master to perform this change, I suggest doing it here too